### ISSUE TITLE
Links with unions with more than two members are not properly unwrapped

### ISSUE THREAD

**Issue by scotttrinh** [edgedb/edgedb-js#646]
**Describe the bug**
If a type has a link that is a union type of more than two members, the query builder generates invalid type signatures.

**Reproduction**
Include the code that is causing the error:

```typescript
export namespace $TypeA0x200x7c0x20default {
  export namespace $TypeB0x200x7c0x20default {
    interface TypeC extends std.BaseObject {}
  }
}
...
"default": {
  "TypeC": TypeC;
  "TypeA | default": {
    "TypeB | default": {
    };
  };
};
```

Introduced in #279

**Comment by CSFlorin** [comment]
Thanks for filing this @scotttrinh was just coming to check if it had been or else I was going to.

**Comment by scotttrinh** [comment]
Fixed in #669 

Released as @edgedb/generate@0.3.1
